### PR TITLE
Checkbox support!

### PR DESCRIPTION
**Pull request by MrArca9**
Keep in mind this was a newb playing with C++ so some things might have to be modified to work correctly. In its current state the code works, but i am unsure if any errors will destroy the program or now.

(This is actually my first pull request, so i don't know if i can select specific files or not.)

### CODE REVIEW

**Comment by notr1ch** [comment]
Thanks for the pull request! The whitespace changes unfortunately make it difficult to isolate exactly which parts have changed, but I think I see how it's done in general. I'll try to replicate this directly.

**Comment by MrArca9** [comment]
No problem!

The main things are as follows:

0) Extend the list view to accept check boxes

1) Get the loading and parsing of the servers attached to the check box clicks. That way when the user clicks on the boxes it auto fills the list

2) when bandwidth testing, pull the item and see if it's checked. If so, test, if not then continue.

File: main.cpp [hunk 1]
```diff
@@ -1529,133 +1285,143 @@ VOID SaveSettings(VOID)
...
-
-		case WM_COMMAND :
-			switch (LOWORD(wParam))
+	}
+	break;
+	case WM_COMMAND:
+		switch (LOWORD(wParam))
+		{
+			//A command was fired, which one?
+		case IDC_US:
```
Comment (MrArca9) [inline main.cpp]: here we check if the user has clicked one of the boxes, and run the function to populate the list.

File: main.cpp [hunk 2]
```diff
@@ -124,12 +128,13 @@ static const AVal av_NetStream_Play_Start = AVC("NetStream.Play.Start");
 static const AVal av_Started_playing = AVC("Started playing");
 static const AVal av_NetStream_Play_Stop = AVC("NetStream.Play.Stop");
 static const AVal av_Stopped_playing = AVC("Stopped playing");
-
 static const AVal av_OBSVersion = AVC("TwitchTest/1.3");
 static const AVal av_setDataFrame = AVC("@setDataFrame");
+static json_t *ingests;
+static int *indexOrder = NULL;
```
Comment (MrArca9) [inline main.cpp]: Global variables for sharing between the bandwidth test and populate list functions.

File: main.cpp [hunk 3]
```diff
@@ -369,10 +374,10 @@ void ProcMainInit (HWND hDlg)
 
 	SendDlgItemMessage(hDlg, IDC_LIST, LVM_SETEXTENDEDLISTVIEWSTYLE, (WPARAM)0, (LPARAM)LVS_EX_FULLROWSELECT | LVS_EX_DOUBLEBUFFER);
 
-	SendDlgItemMessage(hDlg, IDC_EU, BM_SETCHECK, BST_CHECKED, 0);
-	SendDlgItemMessage(hDlg, IDC_US, BM_SETCHECK, BST_CHECKED, 0);
-	SendDlgItemMessage(hDlg, IDC_ASIA, BM_SETCHECK, BST_CHECKED, 0);
-	SendDlgItemMessage(hDlg, IDC_OTHER, BM_SETCHECK, BST_CHECKED, 0);
```
Comment (MrArca9) [inline main.cpp]: Make all boxes unchecked by default, so users can select one or more at run time.

File: main.cpp [hunk 4]
```diff
@@ -424,11 +429,12 @@ void ProcMainInit (HWND hDlg)
 
 	col.mask = LVCF_FMT | LVCF_WIDTH | LVCF_TEXT;
 	col.fmt = LVCFMT_LEFT;
-
-	col.pszText = L"Server";	 col.iOrder = 0; col.cx = 150; ListView_InsertColumn(GetDlgItem(hDlg, IDC_LIST), 0, &col);
-	col.pszText = L"Bandwidth";  col.iOrder = 1; col.cx = 80; ListView_InsertColumn(GetDlgItem(hDlg, IDC_LIST), 1, &col);
-	col.pszText = L"RTT";		col.iOrder = 2; col.cx = 60; ListView_InsertColumn(GetDlgItem(hDlg, IDC_LIST), 2, &col);
-	col.pszText = L"Quality";	col.iOrder = 3; col.cx = 60; ListView_InsertColumn(GetDlgItem(hDlg, IDC_LIST), 3, &col);
+	ListView_SetExtendedListViewStyle(GetDlgItem(hDlg, IDC_LIST), LVS_EX_CHECKBOXES | LVS_EX_FULLROWSELECT);
```
Comment (MrArca9) [inline main.cpp]: extend the listview class to support using check boxes.

File: main.cpp [hunk 5]
```diff
@@ -1176,25 +879,77 @@ unsigned int __stdcall BandwidthTest(void *arg)
 		SendDlgItemMessage(hwndMain, IDC_LIST, LVM_INSERTITEM, 0, (LPARAM)(const LPLVITEM)&lvItem);
 		index++;
 	}
-
-	index = 0;
+}
+
+
+unsigned int __stdcall BandwidthTest(void *arg)
```
Comment (MrArca9) [inline main.cpp]: the new bandwidth class that doesn't populate the list.

**Comment by MrArca9** [comment]
Hopefully these comments clear up the process a little bit.

**Comment by notr1ch** [comment]
This was implemented a while back, forgot to check this PR !

**Pull request closed without merging**